src/preferences/beatdetectionsettings.h

```diff
@@ -19,6 +19,9 @@
 #define BPM_REANALYZE_WHEN_SETTINGS_CHANGE "ReanalyzeWhenSettingsChange"
 #define BPM_FAST_ANALYSIS_ENABLED "FastAnalysisEnabled"
 
+#define BPM_ENABLE_IRONING "BeatDetectionEnableIroning"
+#define BPM_ENABLE_ARRYTHMIC_REGIONS_REMOVAL "BeatDetectionEnableArrythmicRemoval"
```

Comment:
Why are these preferences? What is the use case for not using them?

Reply:
Enable arrhythmic may remove a intentional musical change and be unwanted, not sure about the ironing I can`t think of why not use it actually, maybe on some super edge cases it will perform worse than the raw fluctuations..

Reply:
Have you ever actually had that be a problem? I have not.

Reply:
We have only tested a few tracks. For a good community feedback we need a way to easily revert back to the 2.3 algorithm. That's the main reason for this option. 
If it turns out that there is no other use case than that we can remove the preferences option. 
I think we can also remove the pure constant option in turn, but let us collect opinions first.   

The ironing comes with the cost to not follow the musician. 
We have discussed to put this into an abstract strength parameter that tweaks the internal parameter for a certain type of track. 

For now we are quite happy with the defaults though.

Reply:
I am opposed to adding more options without good reason. The ultimate goal here is to not require any manual intervention from the user.

> For a good community feedback we need a way to easily revert back to the 2.3 algorithm.

Huh? Why? Few people even knew it existed before.

Reply:
Consider this an intermediate state until we have the experience which preferences options we really need. 

Just after merge we need a way to revert back to the old behavior, in case we have messed up something for some tracks.

Reply:
If we add a TODO to our release checklist and remove that option before releasing 2.4 I think it's okay to add it now.

Reply:
Nobody has provided a good reason why these should ever be disabled. Please remove them.

Reply:
I agree with @daschuer that we should have the preferences at least while this is in an "experimental" state.
It allows to simply fallback to the old behavior if we unaware mess something up.

Reply:
If you want to keep these, please provide an example track where the outcome is worse with them. Otherwise we are just adding more clutter to the preferences.

Reply:
The issues is that we don't have the overview yet. We need an easy way for the average user to switch between old an new behavior to compare the results.  

The "The Learning Process" track is a good example where the arrhythmic detection fails for instance. The vocals are detected as beats. If we now improve the algorithm using this track, we make it worse for other tracks where beats are detected as vocals. In this case user may need to disable the arrhythmic detection.

Music is so different that we IMHO need to provide different strategies even in the release version, reflected in preferences option. These will finally be part of a release version of Mixxx, but will look totally different, without turning implementation details to the outside. 

This is just an in-progress state.

Reply:
I think the main use case for the preferences is developer wise than anything else. 
Instead of the need to recompile the code to use/see different results a preference makes it so much easier.

Reply:
The preferences are not a place for developer tools, they're for users.

Reply:
Yes, right. We have already explained that we will change them according to our growing experience during user tests. 
So what is your issue?

Reply:
In that case I agree with @Holzhaus. We can merge them, but they should be removed before we release 2.4, probably before releasing 2.4 beta.

Reply:
Sounds good to me!